### ISSUE TITLE
Passengers don't see other entities get spawned?

### ISSUE THREAD

**Issue by emortaldev** [Minestom/Minestom#705]
Players who are riding another entity (e.g. armourstand) will not see another player start riding that entity

I have made a minimal example

```java
public static void main(String[] args) {
    MinecraftServer server = MinecraftServer.init();

    InstanceManager instanceManager = MinecraftServer.getInstanceManager();
    InstanceContainer instance = instanceManager.createInstanceContainer();

    instance.setChunkGenerator(new Generator());

    GlobalEventHandler global = MinecraftServer.getGlobalEventHandler();

    global.addListener(PlayerLoginEvent.class, (e) -> {
        final Player player = e.getPlayer();
        e.setSpawningInstance(instance);
        player.setGameMode(GameMode.CREATIVE);
        player.setRespawnPoint(new Pos(5, 1, 5));
    });

    global.addListener(PlayerPacketEvent.class, (e) -> {
        final ClientPacket packet = e.getPacket();

        if (packet instanceof ClientSteerVehiclePacket) {
            if (((ClientSteerVehiclePacket)packet).flags() == 2) {
                final Player player = e.getPlayer();
                final Entity vehicle = player.getVehicle();
                if (vehicle != null) {
                    vehicle.removePassenger(player);
                    vehicle.remove();
                }
            }
        }
    });

    MinecraftServer.getCommandManager().register(new SitCommand());

    server.start("0.0.0.0", 25565);
}

public static class SitCommand extends Command {
    public SitCommand() {
        super("sit");

        setDefaultExecutor((sender, context) -> {
            final Player player = (Player) sender;
            final Entity entity = new Entity(EntityType.ARMOR_STAND);

            entity.setInstance(player.getInstance(), player.getPosition()).thenRun(() -> {
                entity.addPassenger(player);
            });
        });
    }
}

public static class Generator implements ChunkGenerator {
    @Override
    public void generateChunkData(@NotNull ChunkBatch batch, int chunkX, int chunkZ) {
        for (int x = 0; x < 16; x++) {
            for (int z = 0; z < 16; z++) {
                int actualX = chunkX * 16 + x;
                int actualZ = chunkZ * 16 + z;
                batch.setBlock(actualX, 0, actualZ, Block.GRASS_BLOCK);
            }

        }
    }

    @Override
    public @Nullable List<ChunkPopulator> getPopulators() {
        return null;
    }
}
```

**Comment by emortaldev** [comment]
https://cdn.discordapp.com/attachments/785295028285079613/945072440416018482/seatingvid.mp4

**Comment by TheMode** [comment]
FIxed by dc90307e80c296eee9cd33c40d84cebd0d3c2262
